database/index.go

```diff
@@ -21,16 +21,33 @@ var (
 )
 
 // An Index associates encoded values with keys.
-// It is sorted by value following the lexicographic order.
+//
+// The association is performed by encoding the values in a binary format that preserve
+// ordering when compared lexicographically. For the implementation, see the binarysort
+// package and the document.ValueEncoder.
+//
+// When the index is composite, the values are wrapped into a document.Array before
+// being encoded.
 type Index struct {
 	Info *IndexInfo
 
 	tx        engine.Transaction
 	storeName []byte
 }
 
-// NewIndex creates an index that associates a value with a list of keys.
+// NewIndex creates an index that associates values with a list of keys.
 func NewIndex(tx engine.Transaction, idxName string, opts *IndexInfo) *Index {
+	if opts == nil {
+		opts = &IndexInfo{
+			Types: []document.ValueType{0},
```

Comment:
🕐 We should define a type for the zero value of ValueTypes

Reply:
🕐 What about: 
- having the zero value always invalid and never being equal to any type
- having an additional type `any` to use where we previously used the zero value (max int?)